### PR TITLE
Add libheif - a HEIF and AVIF (AV1 Image File Format) file format decoder and encoder

### PR DESCRIPTION
**Pull request by frmdstryr**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by frmdstryr** [comment]
`@conda-forge/help-c-cpp`, ready for review!`

File: recipes/libheif/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  version: {{ version }}
+
+source:
+  url: https://github.com/strukturag/libheif/releases/download/v{{ version }}/libheif-{{ version }}.tar.gz
+  sha256: d016905e247d6952cd7ee4f9b90957350b6a6caa466bc76fdfe6eb302b6d088c
+
+build:
+  number: 0
+  run_exports:
+    - {{ pin_subpackage('libheif', max_pin='x.x.x') }}
```
Comment (hmaarrfk) [inline recipes/libheif/meta.yaml:15]: is it really that unstable in terms of ABI?

File: recipes/libheif/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    - libavif
+    - dav1d
+    - aom
+    - x265
+  run:
+    - libde265
+    - libavif
+    - dav1d
+    - aom
+    - x265
```
Comment (hmaarrfk) [inline recipes/libheif/meta.yaml:33]:
```suggestion
```

All of the packages above have run_exports. These runtime dependencies will be automatically added.

**Review by hmaarrfk — COMMENTED** [review]
just a few small comments.

**Comment by hmaarrfk** [comment]
Do you care about GPL vs LPGL licensing? If so, you may be interested in following something like https://github.com/conda-forge/ffmpeg-feedstock/pull/138

I'm happy to merge this as is if you want to proceed. Recipe looks good.

**Comment by frmdstryr** [comment]
I don't care but if it's not too difficult let there be a choice it might be good to add that. 

Could you suggest the change needed?

**Comment by hmaarrfk** [comment]
The changes are similar to those in the PR linked.

Specifically, the changes in the 
https://github.com/conda-forge/ffmpeg-feedstock/pull/138/files#diff-ff61408cdc05bc9667deeadb55e4aaceb1371972076b6bf6934f9008920f2bd2R1

Create a file: `conda_build_config.yaml` in the recipe directory with the contents

```
license_family:
  - gpl
  - lgpl
```

In the recipe, add some jinja syntax that will give the GPL'ed version of the recipe a higher priority:
```
{%- set number = 4 %}
{%- if license_family == 'gpl' %}
{%- set number = number + 100 %}
{%- endif %}

build:
  number: {{ number }}
  string: {{ license_family }}_h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}
```

Then conditionally require the the package `x265` with:

```
    {% if license_family == 'gpl' %}
    - x265
    {% endif %}
```

I think that should do it even in staged recipes.

**Comment by hmaarrfk** [comment]
Its really strange that x265 makes up the hash, but isn't part of the package. I think it is OK! Great job!

**Comment by frmdstryr** [comment]
Thanks!

**Pull request merged**